dora/src/cannon/codegen.rs

```diff
@@ -938,6 +990,191 @@ where
             self.asm.bind_label_to(jump.label, offset);
         }
     }
+
+    fn emit_new_object(&mut self, dest: Register, class_def_id: ClassDefId) {
+        assert_eq!(self.bytecode.register_type(dest), BytecodeType::Ptr);
+
+        let cls = self.vm.class_defs.idx(class_def_id);
+        let cls = cls.read();
+
+        self.asm.emit_comment(Comment::Alloc(class_def_id));
+
+        let alloc_size = match cls.size {
```

Comment:
I think we should always have fixed-size allocations in NewObject (we need an additional bytecode for allocation arrays).

Reply:
Changed the `unimplemented` in the catchall arm to `unreachable`. An additional assert is not necessary.